go/rpc/encrypted_client.go

```diff
@@ -177,19 +179,38 @@ func (c *EncRPCClient) Subscribe(ctx context.Context, namespace string, ch inter
 	return subscription, nil
 }
 
-func (c *EncRPCClient) createAuthenticatedLogSubscription() (*common.LogSubscription, error) {
+func (c *EncRPCClient) createAuthenticatedLogSubscription(args []interface{}) (*common.LogSubscription, error) {
 	accountSignature, err := crypto.Sign(c.Account().Hash().Bytes(), c.viewingKey.PrivateKey.ExportECDSA())
 	if err != nil {
 		return nil, fmt.Errorf("could not sign account address to authenticate subscription. Cause: %w", err)
 	}
-	logSubscription := common.LogSubscription{
+
+	logSubscription := &common.LogSubscription{
 		SubscriptionAccount: &common.SubscriptionAccount{
 			Account:   c.Account(),
 			Signature: &accountSignature,
 		},
-		// TODO - #453 - Add the incoming filters.FilterCriteria to the common.LogSubscription.
 	}
-	return &logSubscription, nil
+
+	// If there are less than two arguments, it means no filter criteria was passed.
+	if len(args) < 2 {
+		logSubscription.Filter = &filters.FilterCriteria{}
+	} else {
+		filterCriteriaJSON, err := json.Marshal(args[1])
+		if err != nil {
+			return nil, fmt.Errorf("could not marshall filter criteria to JSON. Cause: %w", err)
+		}
+
+		filterCriteria := &filters.FilterCriteria{}
+		err = json.Unmarshal(filterCriteriaJSON, &filterCriteria)
```

Comment:
How comes we have to marshal and then unmarshal here? Guess casting doesn't work for some reason? Maybe worth a comment to avoid confusion/people trying to get rid of it without realising why it's needed.

Reply:
Yeah, I'll add one. Basically, we get the object as a map, and we need it as JSON to map it to a filter criteria